src/compiler/crystal/macros/methods.cr

```diff
@@ -1084,6 +1084,39 @@ module Crystal
     end
   end
 
+  class ProcNotation
+    def interpret(method, args, block, interpreter)
+      case method
+      when "inputs"
+        interpret_argless_method(method, args) { ArrayLiteral.new(@inputs || [] of ASTNode) }
+      when "output"
+        interpret_argless_method(method, args) { @output || NilLiteral.new }
+      else
+        super
+      end
+    end
+  end
+
+  class ProcLiteral
+    def interpret(method, args, block, interpreter)
+      # These must be all the methods on Def, ecluding `name`.
+      if {
+        "args",
+        "splat_index",
+        "double_splat",
+        "block_arg",
+        "return_type",
+        "body",
+        "receiver",
+        "visibility"
```

Comment:
Now that I see the list, I'm pretty sure all of these don't make sense in a ProcLiteral: `splat_index`, `double_splat`, `block_arg`, `return_type`, `visibility`. So it seems only `receiver`, `args` and `body` make sense. And please use `case ... when` instead of `{..}.includes?(...)`. Thank you!

Reply:
Amended to just include those methods. I'm not super familiar with Crystal yet so didn't catch that those didn't make sense!